0007-adopt-typescript-in-backend.md

```diff
@@ -0,0 +1,81 @@
+# Adopt Typescript in our Backend
+
+* Status: proposed 
+<!-- {proposed | rejected | accepted | deprecated | … | superseded by [ADR-0005](0005-example.md)} --> <!-- optional -->
+* Deciders: ZavenArra Daniel Dadiorchen Nick Aron
+<!-- Find deciders here: https://github.com/orgs/Greenstand/people  -->
+* Date: 2022-06-02 <!-- optional -->
+
+Technical Story: {description | ticket/issue URL} <!-- optional -->
+
+## Context and Problem Statement
+
+With the backend getting more and more complex, we need to adopt Typescript to make it easier to maintain, code and read.
+
+## Decision Drivers <!-- optional -->
+
+* The complexity of the backend is growing, a weak type of language is becoming too hard to maintain the interface between: function, class, file.
+* The code productivity.
+* The code quality.
+
+## Considered Options
+
```

Comment:
We could consider compiled languages too like Kotlin, but I understand Typescript is a natural progression from javascript.